content/test/integration/controller/deployment-pagination.spec.ts

```diff
@@ -0,0 +1,323 @@
+    if (a.entity.id.toLowerCase() < b.entity.id.toLowerCase()) {
+      E1 = a
+      E2 = b
+    } else {
+      E1 = b
+      E2 = a
+    }
+    const laterTimestamp = Date.now()
+    E3 = await buildDeployData([P3], { type, timestamp: laterTimestamp, metadata: 'metadata3' })
+  })
+
+  it('When there is no next page then the link is undefined', async () => {
+    // Deploy E1, E2 and E3 in that order
+    await deploy(E1, E2, E3)
+
+    const actualDeployments = await fetchDeployments({
+      limit: 3
+    })
+
+    expect(actualDeployments.deployments.length).toBe(3)
+    const nextLink = actualDeployments.pagination.next
+    expect(nextLink).toBeUndefined()
+  })
+
+  it('When local timestamp filter is set, then only toLocalTimestamp is modified in next ', async () => {
+    // Deploy E1, E2 and E3 in that order
+    const [E1Timestamp, E2Timestamp, E3Timestamp] = await deploy(E1, E2, E3)
+
+    const actualDeployments = await fetchDeployments({
+      limit: 2,
+      filters: { fromLocalTimestamp: E1Timestamp, toLocalTimestamp: E3Timestamp }
+    })
+
+    expect(actualDeployments.deployments.length).toBe(2)
+    const nextLink = actualDeployments.pagination.next
+    expect(nextLink).toContain(`from=${E1Timestamp}`)
+    expect(nextLink).toContain(`to=${E2Timestamp}`)
+    expect(nextLink).toContain(`lastId=${E2.entity.id}`)
+  })
+
+  it('When local timestamp filter is set with asc order, then only from is modified in next', async () => {
+    // Deploy E1, E2 and E3 in that order
+    const [E1Timestamp, E2Timestamp, E3Timestamp] = await deploy(E1, E2, E3)
+
+    const actualDeployments = await fetchDeployments({
+      limit: 2,
+      sortBy: {
+        order: SortingOrder.ASCENDING
+      },
+      filters: { fromLocalTimestamp: E1Timestamp, toLocalTimestamp: E3Timestamp }
+    })
+
+    expect(actualDeployments.deployments.length).toBe(2)
+    const nextLink = actualDeployments.pagination.next
+    expect(nextLink).toContain(`from=${E2Timestamp}`)
+    expect(nextLink).toContain(`to=${E3Timestamp}`)
+    expect(nextLink).toContain(`lastId=${E2.entity.id}`)
+  })
+
+  it('When limit is set, then in next it persists', async () => {
+    // Deploy E1, E2 and E3 in that order
+    await deploy(E1, E2, E3)
+
+    const actualDeployments = await fetchDeployments({
+      limit: 2
+    })
+
+    expect(actualDeployments.deployments.length).toBe(2)
+    const nextLink = actualDeployments.pagination.next
+    expect(nextLink).toContain('limit=2')
+  })
+
+  it('When order is by entity timestamp, then only to is modified in next', async () => {
+    // Deploy E1, E2 and E3 in that order
+    const [E1Timestamp, , E3Timestamp] = await deploy(E1, E2, E3)
+
+    const actualDeployments = await fetchDeployments({
+      limit: 2,
+      sortBy: {
+        field: SortingField.ENTITY_TIMESTAMP
+      },
+      filters: { fromLocalTimestamp: E1Timestamp, toLocalTimestamp: E3Timestamp }
```

Comment:
Is this valid? Using `fromLocalTimestamp` or `toLocalTimestamp` when sorting with entity timestamp? Shouldn't it be `from` & `to`?

Reply:
True, I forgot to fail on that case 😄